src/tests/java/htsjdk/samtools/filter/JavascriptSamRecordFilterTest.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2015 Pierre Lindenbaum @yokofakun Institut du Thorax - Nantes - France
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package htsjdk.samtools.filter;
+
+import htsjdk.samtools.SAMRecordIterator;
+import htsjdk.samtools.SamReader;
+import htsjdk.samtools.SamReaderFactory;
+import htsjdk.samtools.util.CloserUtil;
+
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.File;
+
+/**
+ * @author Pierre Lindenbaum PhD Institut du Thorax - INSERM - Nantes - France
+ */
+
+public class JavascriptSamRecordFilterTest {
+	final File testDir = new File("./testdata/htsjdk/samtools");
+
+	@DataProvider
+	public Object[][] jsData() {
+		return new Object[][] { { "unsorted.sam", "samFilter01.js" }, { "unsorted.sam", "samFilter02.js" }, };
+	}
+
+	@Test(dataProvider = "jsData")
+	public void testJavascriptFilters(final String samFile, final String javascriptFile) {
+		final SamReaderFactory srf = SamReaderFactory.makeDefault();
+		final SamReader samReader = srf.open(new File(testDir, samFile));
+		final JavascriptSamRecordFilter filter = new JavascriptSamRecordFilter(new File(testDir, javascriptFile),
+		        samReader.getFileHeader());
+		final SAMRecordIterator iter = samReader.iterator();
+		while (iter.hasNext()) {
+			filter.filterOut(iter.next());
```

Comment:
it would be good to check that you actually get some records out...do you know how many records you expect out of this iterator? might be nice to assert that..